### PR TITLE
Custom Menu's and M600 support

### PR DESCRIPTION
**Pull request by NNMavy**
This PR adds various custom menu's for LCD users that can be used to:

- preform basic calibration tasks such as Z-Tilt.
- perform filament changes.
- pause and cancel prints (The original octoprint menu was broken and ahs been disabled).

To better support these functions it also modifies a few existing macros and add's a bunch of new ones:

- LOW_TEMP_CHECK checks if the hotend is already set to a specific temperature and heats the hotend to 230 degrees if needed
- PARK sends the nozzle to a safe area for loading/unloading of filament.
- CHANGE_FILAMENT aka M600 can be used to change filament mid print and relies on the existing filament macros.

It removes the TURN_OFF_HEATERS command from the load filament macro in order to keep the nozzle's predefined temperature.

### CODE REVIEW

File: macros.cfg [hunk 1]
```diff
@@ -72,4 +73,56 @@ gcode:
...
+    M117 Parking toolhead...
+    {% if printer.gcode_move.position.z|float < (printer.toolhead.axis_maximum.z - 20) %} ; Check if we are below maximum tool height
+        G91 ; use relative coordinates
+        G1 Z{Z} F7200 ; raise z axis by Z mm
+    {% else %}
+        G90 ; use absolute coordinates
+        G1 Z{printer.toolhead.axis_maximum.z} F7200 ; bring z axis to max height
+    {% endif %}
+    G90 ; use absolute coordinates
+    G1 X{X} Y{Y} F7200 ; move to park position
```
Comment (miklschmidt) [inline macros.cfg:121]: I'd prefer if these could be `X{printer.toolhead.axis_maximum.x / 2} Y{printer.toolhead.axis_maximum.y}` by default. The we can use this macro in START_PRINT, END_PRINT and PAUSE as well. You can use the default() jinja method to check for params instead, and drop the default_parameter_X and Y.

File: templates/v-core-3-printer.template.cfg [hunk 2]
```diff
@@ -140,6 +140,11 @@ pid_Kd: 114
 [include config/macros.cfg]
 [include config/printers/v-core-3/macros.cfg]
 
+### DISPLAY MENUS
+#[include config/menus/calibration.cfg]
+#[include config/menus/filament-change.cfg]
+#[include config/menus/printer.cfg]
+
```
Comment (miklschmidt) [inline templates/v-core-3-printer.template.cfg:147]: I think a config/menus/*.cfg would be good here!

**Review by miklschmidt — COMMENTED** [review]
This is awesome!!

I think we should probably put the menu macro's in `lcd` instead of `menu` to make it more clear what it's for.

Excellent job, looking forward to merging! :)

File: macros.cfg (hunk 1 above)
Comment (NNMavy) [reply macros.cfg]: I set the default to the back rear and modified the CHANGE_FILAMENT macro to use 0,0 for easy of access to the nozzle.

File: templates/v-core-3-printer.template.cfg (hunk 2 above)
Comment (NNMavy) [reply templates/v-core-3-printer.template.cfg:145]: Moved and updated though we might have to change the wildcard back in the future once we add more optional menus.

**Comment by NNMavy** [comment]
All the macros have been tested on my end so it should be good to go for a test drive on your end.

**Comment by miklschmidt** [comment]
Excellent! I'm currently rebuilding the V3, will slap on an LCD screen and give it a go when it's back up and running :)

**Comment by NNMavy** [comment]
Closing in favour of #6

**Pull request closed without merging**